k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/K8sResourceHandler.kt

```diff
@@ -44,10 +44,12 @@ class K8sResourceHandler (
                 // notify change to the k8s vanilla image artifact based
                 // on retrieved data
                 manifest?.let {
-                    val imageString = find(manifest.spec as MutableMap<String, Any?>, IMAGE) as String?
-                    imageString?.let { image ->
-                        log.info("Deployed artifact $image")
-                        notifyArtifactDeployed(r, getTag(image))
+                    if (!it.spec.isNullOrEmpty()) {
+                        val imageString = find(manifest.spec as MutableMap<String, Any?>, IMAGE) as String?
+                        imageString?.let { image ->
+                            log.info("Deployed artifact $image")
+                            notifyArtifactDeployed(r, getTag(image))
+                        }
```

Comment:
wouldn't the following have done it, rather than having to add the if block?

```
val imageString = find(manifest.spec ?: mutableMapOf()) as String?
```

Reply:
Ah good idea. Will change